File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'fdd'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var db = new Tinkerforge.BrickletDualButton(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd
// Don't use device before ipcon is connected

// Register state changed callback
db.on(Tinkerforge.BrickletDualButton.CALLBACK_STATE_CHANGED,
    function(buttonL, buttonR, ledL, ledR) {
        if(buttonL === Tinkerforge.BrickletDualButton.BUTTON_STATE_PRESSED) {
            console.log('Left button pressed');
        }
        else {
            console.log('Left button Released');
        }
        if(buttonR === Tinkerforge.BrickletDualButton.BUTTON_STATE_PRESSED) {
            console.log('Right button pressed');
        }
        else{
            console.log('Right button Released');
        }
        console.log();
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Dual Button Bricklet

	// Callback function for state changed callback
	static void StateChangedCB(BrickletDualButton sender, byte buttonL, byte buttonR,
	                           byte ledL, byte ledR)
	{
		if(buttonL == BrickletDualButton.BUTTON_STATE_PRESSED)
		{
			Console.WriteLine("Left Button: Pressed");
		}
		else if(buttonL == BrickletDualButton.BUTTON_STATE_RELEASED)
		{
			Console.WriteLine("Left Button: Released");
		}

		if(buttonR == BrickletDualButton.BUTTON_STATE_PRESSED)
		{
			Console.WriteLine("Right Button: Pressed");
		}
		else if(buttonR == BrickletDualButton.BUTTON_STATE_RELEASED)
		{
			Console.WriteLine("Right Button: Released");
		}

		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletDualButton db = new BrickletDualButton(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register state changed callback to function StateChangedCB
		db.StateChangedCallback += StateChangedCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
